back/src/Guide/Types/Core.hs

```diff
@@ -74,6 +76,20 @@ instance Aeson.ToJSON Trait where
   toJSON = Aeson.genericToJSON Aeson.defaultOptions {
     Aeson.fieldLabelModifier = over _head toLower . drop (T.length "trait") }
 
+instance Aeson.FromJSON Trait where
+  parseJSON = Aeson.withObject "Trait" $ \o -> do
+    traitUid     <- o Aeson..: "uid"
+    content      <- o Aeson..: "content"
```

Comment:
Since `MarkdownInline` has a `FromJSON` instance, you can just write `traitContent <- o Aeson..: "content"`.